### ISSUE TITLE
selector-descendant-combinator-no-non-space

### ISSUE THREAD

**Issue by evilebottnawi** [stylelint/stylelint#1592]
Bad:
```css
.error /* Disallow newline between */
p,
p.explanation {
    color: #fff;
}
```

Good:
```css
.error p,
p.explanation {
    color: #fff;
}
```

**Comment by jeddy3** [comment]
I think what you're wanting to control here is the [whitespace of the whitespace descendent selector](https://drafts.csswg.org/selectors/#descendant-combinators).

We already have:

- `selector-combinator-space-after`
- `selector-combinator-space-after`

To control newlines around non-whitespace combinators like `+`, `>` and `>>`.

So, I think we need a rule to specify *what* whitespace is acceptable for the whitespace descendent combinator. I guess what would be most useful is a rule that only allows a single space, rather than multiple spaces or newlines. I'm not quite sure what we'd call such a thing though... perhaps:

- `selector-whitespace-descendant-combinator-no-non-space`?

We need the `whitespace` bit to distinguish this combinator from the non-whitespace descendant one (i.e. `>>`).

**Comment by davidtheclark** [comment]
Never heard of the >> combinator before. Could you please link to info?

**Comment by jeddy3** [comment]
It's the link in my post... ["whitespace of the whitespace descendent selector"](https://drafts.csswg.org/selectors/#descendant-combinators). It's draft, but we might as well protect ourselves from it.

**Comment by davidtheclark** [comment]
SGTM

**Comment by evilebottnawi** [comment]
@jeddy3 Should we throw error only on newline? or also tab and etc spaced symbols?

**Comment by jeddy3** [comment]
Throw on anything that isn't a *single space*. `*-space` is used consistently that way throughout the rules.

**Comment by evilebottnawi** [comment]
@jeddy3 let's rock after release v7.0.0 :tada:

**Comment by jeddy3** [comment]
Done in https://github.com/stylelint/stylelint/issues/1784

**Comment by evilebottnawi** [comment]
@jeddy3 Thanks for assistance in completing PR :smile:
